### PR TITLE
feat: add support to configure amp story page path through featureConfig

### PR DESCRIPTION
**Pull request by cricket-lover**
https://github.com/quintype/page-builder/issues/2737

This path has to be passed inside featureConfig. It can passed as a `function` or a `string`

```
featureConfig: {
    ampPageBasePath: () => "/amp"
}
```

### CODE REVIEW

**Review by nandap4790 — APPROVED** [review]

File: server/amp/helpers/get-amp-page-base-path.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+function getAmpPageBasePath(opts = {}) {
+  let ampPageBasePath = "/amp/story";
+  if (opts.featureConfig) {
+    const configAmpPath =
+      typeof opts.featureConfig.ampPageBasePath === "function"
+        ? opts.featureConfig.ampPageBasePath()
```
[the remark this answers is not in the document]
Comment (Jeevan-Kishore) [reply server/amp/helpers/get-amp-page-base-path.js:6]: +1

File: server/amp/helpers/get-amp-page-base-path.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        : opts.featureConfig.ampPageBasePath;
+    ampPageBasePath = configAmpPath || ampPageBasePath;
```
Comment (Jeevan-Kishore) [inline server/amp/helpers/get-amp-page-base-path.js:8]: Return if true

File: server/amp/helpers/get-amp-page-base-path.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  }
+
+  return ampPageBasePath;
```
Comment (Jeevan-Kishore) [inline server/amp/helpers/get-amp-page-base-path.js:11]: Add this as default return

**Review by Jeevan-Kishore — CHANGES_REQUESTED** [review]

**Review by VeenaYemmiganur — APPROVED** [review]

**Review by nandap4790 — APPROVED** [review]

File: server/amp/helpers/get-amp-page-base-path.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const get = require("lodash/get");
+
+function getAmpPageBasePath(opts = {}) {
+  let ampPageBasePath = get(opts, ["featureConfig", "ampPageBasePath"], "/amp/story");
+  ampPageBasePath = typeof ampPageBasePath === "function" ? ampPageBasePath() : ampPageBasePath;
```
Comment (ags1773) [inline server/amp/helpers/get-amp-page-base-path.js:5]: You'll need to pass `config` to `ampPageBasePath` function.. Madrid will need config to access pagebuilder config

Madrid code:
Screenshot 2023-11-29 at 11 33 21 AM

**Pull request merged**